### ISSUE TITLE
Mismatched Versions of React w/ CRA

### ISSUE THREAD

**Issue by jasondashwang** [typekev/react-mk#11]
While trying to use this library for an unejected `create-react-app` (using `"react": "^17.0.1"`), I ran into this issue immediately:

```
Uncaught Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons: 
1. You might have mismatching versions of React and the renderer (such as React DOM) 
2. You might be breaking the Rules of Hooks 
3. You might have more than one copy of React in the same app
```

Turns out the [solution](https://github.com/facebook/react/issues/13991#issuecomment-435587809) was to eject the app and add this line to my `webpack.config.js` in order to match the react versions.
```js
alias: {
  react: path.resolve('./node_modules/react')
}
```

Not sure what the fix is but wanted to provide an FYI in case future people run into this.

**Comment by typekev** [comment]
Thank you very much, I really appreciate it, @jasondashwang, I will look into this today.

**Comment by Danktuary** [comment]
Hey @typekev, I also stumbled upon this error tonight on one of my own projects (also a React component library, made with `create-react-app`) and this comment is what solved my issue: https://github.com/facebook/react/issues/13991#issuecomment-718693370

To recap, `react` and `react-dom` should be peer dependencies and not direct/dev dependencies. It looks like you have `react` as a direct dep and `react-dom` as a dev dep here.
And @jasondashwang, in your React app, I believe `react` and `react-dom` should be dev dependencies, if they aren't already.

Hope this helps. 😄

**Comment by typekev** [comment]
Thank you very much for your help @Danktuary, I have just implemented the changes in #12 
Tested on my own site (`react` v16) and on a new CRA installation (`react` v17), seems to have fixed the issue.

**Comment by typekev** [comment]
Thank you again for reporting the issue, @jasondashwang. The fix is live on npm now, published in `v1.1.4`.
If you try out `react-mk` again, please let me know if it works as expected 👍
